src/tools/imgtool/library.cpp

```diff
@@ -128,155 +110,124 @@ void imgtool_library_add(imgtool_library *library, imgtool_get_info get_info)
 		i = 0;
 		do
 		{
-			/* clear out the class */
+			// clear out the class
 			memset(&imgclass, 0, sizeof(imgclass));
 			imgclass.get_info = get_info;
 
-			/* make the class */
+			// make the class
 			result = make_class(i++, &imgclass);
 			if (result)
-				imgtool_library_add_class(library, &imgclass);
+				add_class(&imgclass);
 		}
 		while(result);
 	}
 	else
 	{
-		imgtool_library_add_class(library, &imgclass);
+		add_class(&imgclass);
 	}
 }
 
 
+//-------------------------------------------------
+//  unlink
+//-------------------------------------------------
 
-const imgtool_module *imgtool_library_unlink(imgtool_library *library,
-	const char *module)
+void library::unlink(const char *module_name)
 {
-	imgtool_module *m;
-	imgtool_module **previous;
-	imgtool_module **next;
-
-	for (m = library->first; m; m = m->next)
-	{
-		if (!core_stricmp(m->name, module))
-		{
-			previous = m->previous ? &m->previous->next : &library->first;
-			next = m->next ? &m->next->previous : &library->last;
-			*previous = m->next;
-			*next = m->previous;
-			m->previous = nullptr;
-			m->next = nullptr;
-			return m;
-		}
-	}
-	return nullptr;
+	modulelist::iterator iter = find(module_name);
```

Comment:
You could make this `const` since you don't assign to `iter` after initialisation (you just can't make it `const_iterator`).